zipkin-lens/src/components/DiscoverPage/TraceSummaryRow.tsx

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2015-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import {
+  Box,
+  Collapse,
+  IconButton,
+  TableCell,
+  TableRow,
+  Typography,
+  useTheme,
+  Button,
+} from '@material-ui/core';
+import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
+import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp';
+import moment from 'moment';
+import React, { useCallback, useMemo, useState } from 'react';
+import { Link } from 'react-router-dom';
+import {
+  Bar,
+  BarChart,
+  CartesianGrid,
+  LabelList,
+  ResponsiveContainer,
+  Tooltip,
+  XAxis,
+  YAxis,
+} from 'recharts';
+import styled from 'styled-components';
+
+import { selectColorByInfoClass, selectServiceColor } from '../../colors';
+import TraceSummary from '../../models/TraceSummary';
+import { formatDuration } from '../../util/timestamp';
+
+const renderCustomizedLabel = (props: any) => {
+  const { x, y, width, value } = props;
+  const radius = 10;
+
+  return (
+    <g>
+      <circle
+        cx={x + width / 2}
+        cy={y - radius}
+        r={radius}
+        fill={selectServiceColor(value)}
+      />
+      <text
+        x={x + width / 2}
+        y={y - radius}
+        fill="#fff"
+        textAnchor="middle"
+        dominantBaseline="middle"
+      >
+        {value[0].toUpperCase()}
+      </text>
+    </g>
+  );
+};
+
+interface TraceSummaryRowProps {
+  traceSummary: TraceSummary;
+}
+
+const TraceSummaryRow: React.FC<TraceSummaryRowProps> = ({ traceSummary }) => {
+  const theme = useTheme();
+  const [open, setOpen] = useState(false);
+  const handleOpenButtonClick = useCallback(() => {
```

Comment:
Believe you may be able to use `useReducer` instead of `useState` + `useCallback` since you only toggle.